capped_file.go

```diff
@@ -0,0 +1,195 @@
+// bagdb: Simple datastorage
+// Copyright 2021 billy authors
+// SPDX-License-Identifier: BSD-3-Clause
+
+package billy
+
+import (
+	"fmt"
+	"os"
+)
+
+// cappedFile has an API-surface as if it were one file, but maps
+// to a set of files on disk. These files are all capped in size
+// to maxFileSize.
+type cappedFile struct {
+	cap   uint64
+	files []*os.File
+}
+
+// newCappedFile creates a cappedFile
+func newCappedFile(basename string, nFiles int, cap uint64, readonly bool) (*cappedFile, error) {
+	flags := os.O_RDWR | os.O_CREATE
+	if readonly {
+		flags = os.O_RDONLY
+	}
+	var files []*os.File
+	for i := 0; i < nFiles; i++ {
+		var (
+			f   *os.File
+			err error
+		)
+		if i == 0 {
```

Comment:
Should this perhaps be `if nFiles == 1`? That way for single files I won't have the strange suffit, but for capped files, I will have a proper `.cap.0` suffix for the first file too.

I guess the catch is if someone sets blobpool sizes below 2 GB and then above, ten they will flip-flop between the 2 naming schemes.

Any particular reason for supporting the "old" scheme? Can't we just always have "basename.%d"?

Reply:
My reasoning was that in most cases, there's only ever going to be one file used. And in that case, the one and only file (the zeroth file) could be just the basename. 
Only if it does go above would we start adding suffixes. So in the 'normal case', there would be zero difference between a system using a capped file and one not using capped files.